lib/puppet/util/execution.rb

```diff
@@ -186,6 +189,11 @@ def self.execute(command, options = NoOptionsSpecified)
 
     null_file = Puppet.features.microsoft_windows? ? 'NUL' : '/dev/null'
 
+    cwd = options[:cwd]
+    if cwd && ! Puppet::FileSystem.directory?(cwd)
```

Comment:
I think we can modify `Puppet::FileSystem.directory?` as part of `PUP-9150` so that this check can still remain at the top-level.